### PR TITLE
Simple longest common subsequence solver

### PR DESCRIPTION
**Pull request by wplohrmann**
This will be useful for comparing iterables later on.

### CODE REVIEW

File: dast/lcs.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+"""
+Longest Common Subsequence problem
+"""
+from collections import defaultdict
```
Comment (casellimarco) [inline dast/lcs.py:4]: :heart_eyes:

File: dast/lcs.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    """
+    x_in_lcs = []
+    n = 0
+    for element in x:
+        if element == lcs[n]:
+            x_in_lcs.append(True)
+            n += 1
+        else:
+            x_in_lcs.append(False)
+
```
Comment (casellimarco) [inline dast/lcs.py]: Should assert n==len(x_in_lcs) or maybe is convenient to keep it flexible?

File: dast/lcs.py [hunk 3]
```diff
@@ -4,43 +4,43 @@
...
+            if x == y:
+                sub_problems[i + 1, j + 1] = sub_problems[i, j] + [x]
             else:
                 sub_problems[i + 1, j + 1] = max(
                     sub_problems[i, j + 1],
                     sub_problems[i + 1, j],
                     key=len)
 
-    lcs = sub_problems[num_x, num_y]
+    lcs = sub_problems[len(xs), len(ys)]
```
Comment (wplohrmann) [inline dast/lcs.py]: In general `Iterable`s do not support `len`, so maybe we should just change the signature to `List` instead?

File: dast/lcs.py (hunk 2 above)
Comment (wplohrmann) [reply dast/lcs.py]: Shouldn't it be `len(x_in_lcs) == len(xs)`?

File: dast/lcs.py (hunk 3 above)
Comment (casellimarco) [reply dast/lcs.py]: Indeed I fixed it in the follow up commit, we can even use i+1, j+1, but it would not be that explicit

Comment (casellimarco) [reply dast/lcs.py]: The diff looks funny though, still has come commits that are already in main

File: dast/lcs.py (hunk 2 above)
Comment (casellimarco) [reply dast/lcs.py]: That's granted, no?
Ah sorry my bad, I meant 'n==len(lcs)'
I was wondering if we want to enforce lcs to be a substring of xs

Comment (casellimarco) [reply dast/lcs.py]: An input like ("a", ["b"]) returns [False], which makes sense in general but maybe we should raise?

Comment (casellimarco) [reply dast/lcs.py]: I am also fine without raising, just wanted to discuss it

**Review by casellimarco — APPROVED** [review]
Have a look at the comment regarding `sequence_in_lcs` then feel free to merge

**Pull request merged**